### DISCUSSION TITLE
Array of jsons... not quite working?

### DISCUSSION THREAD

**Discussion by Dariusz1989** [nlohmann/json#3217] [Q&A] [answered]
Hey

Here is my code:

```
    nlohmann::json item::data() {
        std::vector<nlohmann::json> ruleList(mEntryList.size());
        int x = 0;
        for (auto &item: mEntryList) {
            ruleList[x] = item->data();
            x++;
        }
      return {ruleList};
```
That seems to not work somehow, I tried with iterators to ruleList.begin()/end(). Same issue.

Seems like I Have to do this >
```
  nlohmann::json d;
        d["a"] = ruleList;
        return d["a"];
```
To make it.. work?
Any hints?
TIA

**Comment by nlohmann** [comment, marked as answer]
It seems you need to don't want to return `{ruleList}`, but just `ruleList`.

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?

**Reply by Dariusz1989** [reply]
Shish, sorry I haven't had time to test it O.O, But if u say it's all I need then I'm sure it will work :- D Thank youu!!!
